lib/events/adapters/aws_sns.ex

```diff
@@ -5,20 +5,16 @@ defmodule ElixirTools.Events.Adapters.AwsSns do
 
   require Logger
 
-  alias ElixirTools.Events.Event
-
   @behaviour ElixirTools.Events.Publisher
 
   @typep publish_opt ::
            {:aws_module, module}
            | {:sns_module, module}
-           | {:uuid_module, module}
            | {:group, String.t()}
            | {:topic, String.t()}
-           | {:timex_module, module}
 
   @impl true
-  @spec publish(Event.t(), [publish_opt]) :: :ok | {:error, term}
+  @spec publish(map, [publish_opt]) :: :ok | {:error, term}
```

Comment:
My suggestion is not to create two schema modules, which would make over complicated the event solution. 

Current Event schema should be migrated to the event structure after having added the envelope. This is transparent for clients as event is never directly manipulated, but directly piped to `Event.publish/2` method.

When done, this spec could be moved back to `@spec publish(Event.t(), [publish_opt]) :: :ok | {:error, term}`

Reply:
:thinking: in this case how specs of `EventHandler.create | publish` will looks like?

Reply:
They should look exactly the same:

```elixir
@spec create(event_name, payload, event_id_seed, [create_optional]) :: Event.t()
```

as the `Event.t()` structure should be modified to match with an event such us:

```elixir
%Event{
        action: "TEST_EVENT",
        group: "EVENT_GROUP",
        id: "55989ea4-d947-4f16-b8c8-e0e888facff4",
        occurred_at: "2015-06-29T05:55:55Z",
        payload: %{},
        version: "1.0.0"
      }
```

For th external world, in terms of specs should be transparent. The only difference would be the structure of the event, The event will be, after created, the final event (with the "envelope").

JSON schema validation could even be done here, before publish call. This way, JSON schema validation would be done with the rest of validations that are currently being done.